app/helpers/payment_details_helper.rb

```diff
@@ -19,33 +22,37 @@ def link_to_payment_details(text, options = {})
   end
 
   def payment_details_path(merchant_account = site_account, url_params = {})
-    return "" if ["bogus" ,""].include?(merchant_account.payment_gateway_type.to_s)
-    named_route = [:admin, :account, merchant_account.payment_gateway_type]
+    payment_gateway_type = merchant_account.payment_gateway_type
+    return "" if ["bogus",""].include?(payment_gateway_type.to_s)
+
+    named_route = [:admin, :account, payment_gateway_type]
     polymorphic_path(named_route, url_params)
   end
 
   def edit_payment_details(merchant_account = site_account)
-    return "" if ["bogus" ,""].include?(merchant_account.payment_gateway_type.to_s)
-    polymorphic_url([:edit, :admin, :account, merchant_account.payment_gateway_type])
+    payment_gateway_type = merchant_account.payment_gateway_type
+    return "" if ["bogus",""].include?(payment_gateway_type.to_s)
+
+    polymorphic_url([:edit, :admin, :account, payment_gateway_type])
   end
 
+  # This smells of :reek:FeatureEnvy but it shouldn't
   def merchant_countries
-    ActiveMerchant::Country::COUNTRIES.map{|c| [c[:name], c[:alpha2]] }
+    @merchant_countries ||= ActiveMerchant::Country::COUNTRIES.map { |country| [country[:name], country[:alpha2]] }.uniq
```

Comment:
Is this `uniq` fixing a BUG where we have repeated countries?

Reply:
Ah, forgot to mention that... yes. Right now Romania is duplicated and shows an error in the browser console (because a select can't have 2 equal options).

Reply:
Interesting. Turns out the ISO 3166-1 alpha 3 country code for Romania changed from "ROM" to "ROU" in 2002.

Reply:
That's a mistake, we should have Bulgaria twice instead of Romania!